src/extension.ts

```diff
@@ -183,43 +191,50 @@ export function activate(context: vscode.ExtensionContext) {
 			return;
 		}
 
+		const filesToRemove: string[] = [];
 		const clipboardContent = await vscode.env.clipboard.readText();
 		const clipboard = await writeFileOnDisk(clipboardContent);
+		filesToRemove.push(clipboard);
 
 		//by default compare clipboard against current file
 		let sameContent;
 		let current = editor.document.fileName;
 		const selection = editor.selection;
-		if (!selection.isEmpty) { 
+		if (!selection.isEmpty) {
 			//compare against current selection
 			const editorContent = editor.document.getText(selection);
 			current = await writeFileOnDisk(editorContent);
+			filesToRemove.push(current);
 			sameContent = await areFilesEqual([current, clipboard]);
 		} else if (editor.document.isUntitled) {
 			//compare against untitled file
 			const editorContent = editor.document.getText();
 			current = await writeFileOnDisk(editorContent);
+			filesToRemove.push(current);
 			sameContent = await areFilesEqual([current, clipboard]);
 		} else if (editor.document.isDirty) {
 			//compore against dirty content but invoke meld with current saved file
 			const editorContent = editor.document.getText();
 			const tmpCheck = await writeFileOnDisk(editorContent);
+			filesToRemove.push(tmpCheck);
 			sameContent = await areFilesEqual([tmpCheck, clipboard]);
 		} else {
 			sameContent = await areFilesEqual([current, clipboard]);
 		}
 
 		if (sameContent) {
 			window.showInformationMessage('No difference');
+			cleanupTmpFiles(filesToRemove);
 		} else {
-			showMeld([current, clipboard]);
+			const process = showMeld([current, clipboard]);
+			process.on('exit', () => cleanupTmpFiles(filesToRemove));
```

Comment:
Process could be `undefined`. An `if (process)` should be added.

Reply:
Good point. Wasn't aware of it.

Reply:
OK, I merge the changes as they are and add the if statement afterwards.